zebra-state/src/arbitrary.rs

```diff
@@ -33,6 +33,32 @@ impl Prepare for Arc<Block> {
     }
 }
 
+impl<T> From<T> for ChainTipBlock
+where
+    T: Prepare,
+{
+    fn from(block: T) -> Self {
+        block.prepare().into()
+    }
+}
+
+impl From<PreparedBlock> for ChainTipBlock {
+    fn from(prepared: PreparedBlock) -> Self {
+        let PreparedBlock {
+            block: _,
+            hash,
+            height,
+            new_outputs: _,
+            transaction_hashes,
+        } = prepared;
+        Self {
+            hash,
+            height,
+            transaction_hashes,
+        }
```

Comment:
It's used to avoid cloning the `Arc`s, which is slightly more expensive due to the atomic reference count.

I've also found it useful to explicitly show all the fields, and show which fields we're dropping.